rdmd.d

```diff
@@ -916,3 +900,41 @@ void yap(size_t line = __LINE__, T...)(auto ref T stuff)
     debug stderr.writeln(line, ": ", stuff);
     else stderr.writeln(stuff);
 }
+
+/**
+Used to wrap filesystem operations that should also be logged with the
+yap function. These functions allow the the filenane to be given once
+so the log statement always matches the operation. Using it also guarantees
+you won't forget to include a `yap` alongside any file operation you want
+to be logged.
+*/
+static struct Yap
+{
+    static auto opDispatch(string op, T...)(T args)
+    {
+        static if (op == "exists" ||
+                   op == "timeLastModified" ||
+                   op == "isFile")
+            yap("stat ", args[0]);
+        else static if (op == "rename")
+            yap("mv ", args[0], " ", args[1]);
+        else static if (op == "remove" ||
+                        op == "rmdirRecurse" ||
+                        op == "dirEntries")
+            yap(op, " ", args[0]);
+        else static assert(0, "Yap.opDispath has not implemented " ~ op);
+
+        mixin("return std.file." ~ op ~ "(args);");
+    }
+    static bool existsAsFile(const(char)[] name)
```

Comment:
Did we not agree to drop this one for this PR?

Reply:
If we drop this one then we'll see a bunch of duplicated `stat` calls...

What was dropped was actually implementing 1 file system query, but this one is to consolidate logging so you don't see 2 stat statements in the log.

Reply:
If the intention of this PR is to improve logging accuracy, then we need to be authentic with the goal. If there are two `stat` calls, then there should be two `stat` log lines.

Reply:
See comment above the `Yap` structure.  That's what I mean when I say "improve logging accuracy".  2 file system operations in succession that do not change the behavior of rdmd is just adding noise to the log.